utils/build-windows.bat

```diff
@@ -0,0 +1,314 @@
+:: build-windows.bat
+::
+:: This source file is part of the Swift.org open source project
+::
+:: Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+:: Licensed under Apache License v2.0 with Runtime Library Exception
+::
+:: See https://swift.org/LICENSE.txt for license information
+:: See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+:: REQUIRED ENVIRONMENT VARIABLES
+:: This script requires to be executed inside one Visual Studio command line,
+:: in order for many of the tools and environment variables to be available.
+:: Additionally, it needs the following variables:
+:: - CMAKE_BUILD_TYPE: Kind of build: Release, RelWithDebInfo, Debug.
+:: - PYTHON_HOME: The Python installation directory.
+
+:: REQUIRED PERMISSIONS
+:: Practically, it is easier to be in the Adminstrators group to run the
+:: script, but it should be possible to execute as a normal user.
+:: The user will need permission to write files into the Windows SDK and the
+:: VisualC++ folder.
+
+@echo off
+
+setlocal enableextensions enabledelayedexpansion
+
+set "exitOnError=|| (exit /b)"
+set current_directory=%~dp0
+set current_directory=%current_directory:~0,-1%
+set source_root=%current_directory%\..\..
+
+cd %source_root%
+set source_root=%CD%
```

Comment:
Huh?  Why this dance with the shuffle?  You set it, you cd, you read the value.  Are you trying to emulate `realpath`?

Reply:
Kinda. It avoids `..\..` in every path derived from `source_root` ( and any derived from `build_root` and `install_directory`). The absolute paths were easier to read when I was debugging the script (and easy to copy/paste).

Reply:
I agree.  Though, a comment to explain that would be nice.

Reply:
I will add the comment.